backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -436,6 +465,19 @@ public Response updateEndpoint(@Context SecurityContext sec,
 
         endpointRepository.updateEndpoint(endpoint);
 
+        // Update the secrets in Sources.
+        if (this.featureFlipper.isSourcesUsedAsSecretsBackend()) {
+            // In order to be able to update the secrets in Sources, we need to grab the IDs of these secrets from the
+            // database endpoint, since the client won't be sending those IDs.
+            var incomingEndpointProps = (SourcesSecretable) endpoint.getProperties();
+            var databaseEndpointProps = (SourcesSecretable) ep.getProperties();
+
+            incomingEndpointProps.setBasicAuthenticationSourcesId(databaseEndpointProps.getBasicAuthenticationSourcesId());
+            incomingEndpointProps.setSecretTokenSourcesId(databaseEndpointProps.getSecretTokenSourcesId());
+
+            this.secretUtils.updateSecretsForEndpoint(endpoint);
```

Comment:
What if `endpoint` did not contain any basic auth or secret token data when it was first created, but now it contains that data? I suppose we'll need to create data in Sources.

What if `endpoint` previously contained basic auth or secret token data but they were removed by the user and now everything's empty/null? I suppose we'll need to remove the old data from Sources.

Reply:
> What if endpoint did not contain any basic auth or secret token data when it was first created, but now it contains that data? I suppose we'll need to create data in Sources.

Yeah, agree on this one.

> What if endpoint previously contained basic auth or secret token data but they were removed by the user and now everything's empty/null? I suppose we'll need to remove the old data from Sources.

Good question. I'm guessing that since the clients have to send the entire object to our `PUT /endpoints/{id}`, it is safe for us to assume that when the basic authentication or the secret token are `null`, we can safely delete the secrets?

Reply:
All right, I did implement this behaviour too for the "update" operation!